coderd/coderdtest/authorize.go

```diff
@@ -391,19 +404,21 @@ var _ rbac.PreparedAuthorized = (*fakePreparedAuthorizer)(nil)
 // return the same error as the original FakeAuthorizer.
 type fakePreparedAuthorizer struct {
 	sync.RWMutex
-	Original *FakeAuthorizer
-	Subject  rbac.Subject
-	Action   policy.Action
+	Original               *FakeAuthorizer
+	PreparedRegoAuthorizer rbac.PreparedAuthorized
+	Subject                rbac.Subject
+	Action                 policy.Action
 }
 
 func (f *fakePreparedAuthorizer) Authorize(ctx context.Context, object rbac.Object) error {
 	return f.Original.Authorize(ctx, f.Subject, f.Action, object)
 }
 
-// CompileToSQL returns a compiled version of the authorizer that will work for
-// in memory databases. This fake version will not work against a SQL database.
-func (*fakePreparedAuthorizer) CompileToSQL(_ context.Context, _ regosql.ConvertConfig) (string, error) {
-	return "not a valid sql string", nil
+func (f *fakePreparedAuthorizer) CompileToSQL(ctx context.Context, cfg regosql.ConvertConfig) (string, error) {
+	if f.Original.sqlFilter != "" {
+		return f.Original.sqlFilter, nil
+	}
+	return f.PreparedRegoAuthorizer.CompileToSQL(ctx, cfg)
```

Comment:
I'd remove the real rego authorizer and just always return `true` as the string literal for the response as the default. Would that work?